pkg/cmd/step/git/credentialhelper/step_git_credentialhelper.go

```diff
@@ -0,0 +1,159 @@
+}
+
+var (
+	stepGitCredentialHelperLong = templates.LongDesc(`
+		This pipeline step generates a Git credentials file for the current Git provider secrets
+
+`)
+
+	stepGitCredentialHelperExample = templates.Examples(`
+		# respond to a git credentials request
+		jx step git credentials-helper
+`)
+)
+
+// NewCmdStepGitCredentialHelper creates a new command
+func NewCmdStepGitCredentialHelper(commonOpts *opts.CommonOptions) *cobra.Command {
+	options := StepGitCredentialHelperOptions{
+		StepOptions: step.StepOptions{
+			CommonOptions: commonOpts,
+		},
+	}
+	cmd := &cobra.Command{
+		Use:     "credential-helper",
+		Short:   "Creates a Git credentials helper",
+		Long:    stepGitCredentialHelperLong,
+		Example: stepGitCredentialHelperExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			options.Cmd = cmd
+			options.Args = args
+			err := options.Run()
+			helper.CheckErr(err)
+		},
+	}
+	cmd.Flags().StringVarP(&options.GitKind, "git-kind", "", "", "The git kind. e.g. github, bitbucketserver etc")
+	cmd.Flags().StringVarP(&options.GitHubAppOwner, optionGitHubAppOwner, "g", "", "The owner (organisation or user name) if using GitHub App based tokens")
+
+	return cmd
+}
+
+// Run the main function
+func (o *StepGitCredentialHelperOptions) Run() error {
+	gha, err := o.IsGitHubAppMode()
+	if err != nil {
+		return err
+	}
+
+	if gha {
+		log.Logger().Info("Running in GitHub App mode")
+	} else {
+		log.Logger().Info("Not running in GitHub App mode")
+	}
+
+	var authConfigSvc auth.ConfigService
+	if gha {
+		authConfigSvc, err = o.GitAuthConfigServiceGitHubAppMode(o.GitKind)
+		if err != nil {
+			return errors.Wrap(err, "when creating auth config service using GitAuthConfigServiceGitHubAppMode")
+		}
+	} else {
+		authConfigSvc, err = o.GitAuthConfigService()
+		if err != nil {
+			return errors.Wrap(err, "when creating auth config service using GitAuthConfigService")
+		}
+	}
+
+	credentials, err := o.CreateGitCredentialsFromAuthService(authConfigSvc)
+	if err != nil {
+		return errors.Wrap(err, "creating git credentials")
+	}
+
+	helper, err := credentialhelper.CreateGitCredentialsHelper(os.Stdin, os.Stdout, credentials)
+	if err != nil {
+		return errors.Wrap(err, "unable to create git credential helper")
+	}
+
+	// the credential helper operation (get|store|remove) is passed as last argument to the helper
+	err = helper.Run(os.Args[len(os.Args)-1])
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateGitCredentialsFromAuthService creates the git credentials using the auth config service
+func (o *StepGitCredentialHelperOptions) CreateGitCredentialsFromAuthService(authConfigSvc auth.ConfigService) ([]credentialhelper.GitCredential, error) {
+	var credentialList []credentialhelper.GitCredential
+
+	cfg := authConfigSvc.Config()
+	if cfg == nil {
+		return nil, errors.New("no git auth config found")
+	}
+
+	for _, server := range cfg.Servers {
+		var auths []*auth.UserAuth
+		if o.GitHubAppOwner != "" {
```

Comment:
What's about we remove this and always use _server.User_. Then we build the GitCredentials and pass the UserAuth.GithubAppOwner and use it as path component. When the request comes in we match against the path.